src/kumo-api.ts

```diff
@@ -189,27 +186,43 @@ export class KumoApi {
           label: zoneTable[serial].label,
           zoneTable: zoneTable[serial],
         };
-        this.log.info('Found device. Serial: %s. Label: %s', device.serial, device.label);
-        this.devices.push(device);
+        let existingDeviceIndex: string|number|undefined = undefined;
+        for (const anExistingDeviceIndex in this.devices) {
+          if (this.devices[anExistingDeviceIndex].serial === device.serial) {
+            existingDeviceIndex = anExistingDeviceIndex;
+            break;
+          }
+        }
+
+        if (existingDeviceIndex !== undefined) {
+          this.log.info('Updated existing device. Serial: %s. Label: %s', device.serial, device.label);
+          this.devices[existingDeviceIndex] = device;
```

Comment:
Enhanced parseChildren to check if a device already exists and to update that existing device if it does